enhancements/storage/csi-driver-operator-merge.md

```diff
@@ -0,0 +1,530 @@
+code. For example, the `assets/` directory of each operator is mostly copy-paste from an earlier operator and fixing
+anything there leads to too many PRs. Similarly, `starter.go` is often very similar to the other operators.
+
+The other teams will still be responsible for their platform-specific code, e.g. adding extra controllers
+to sync / modify driver's Secrets, running extra Deployment with a webhook, or even adding extra sidecars to the CSI
+driver like IBM vpc-node-label-updater. We want to reduce copy-paste of the common code, which is easier if the operator
+code is in a single repository.
+
+#### Building and shipping the operators
+
+Right now, ART pipeline builds a separate image for each CSI driver operator:
+
+| Source repository                                                                                                      | ART name*                                    |
+|------------------------------------------------------------------------------------------------------------------------|----------------------------------------------|
+| [openshift/alibaba-disk-csi-driver-operator](https://github.com/openshift/alibaba-disk-csi-driver-operator/)           | ose-alibaba-disk-csi-driver-operator.yml     |
+| [openshift/aws-ebs-csi-driver-operator](https://github.com/openshift/aws-ebs-csi-driver-operator/)                     | ose-aws-ebs-csi-driver-operator.yml          |
+| [openshift/aws-efs-csi-driver-operator](https://github.com/openshift/aws-efs-csi-driver-operator/)                     | ose-aws-efs-csi-driver-operator.yml          |
+| [openshift/azure-disk-csi-driver-operator](https://github.com/openshift/azure-disk-csi-driver-operator/)               | ose-azure-disk-csi-driver-operator.yml       |
+| [openshift/azure-file-csi-driver-operator](https://github.com/openshift/azure-file-csi-driver-operator/)               | azure-file-csi-driver-operator.yml           |
+| [openshift/gcp-pd-csi-driver-operator](https://github.com/openshift/gcp-pd-csi-driver-operator/)                       | ose-gcp-pd-csi-driver-operator.yml           |
+| [openshift/gcp-filestore-csi-driver-operator](https://github.com/openshift/gcp-filestore-csi-driver-operator/)         | ose-gcp-filestore-csi-driver-operator.yml    |
+| [openshift/ibm-powervs-block-csi-driver-operator](https://github.com/openshift/ibm-powervs-block-csi-driver-operator/) | ose-powervs-block-csi-driver-operator.yml    |
+| [openshift/ibm-vpc-block-csi-driver-operator](https://github.com/openshift/ibm-vpc-block-csi-driver-operator/)         | ose-ibm-vpc-block-csi-driver-operator.yml    |
+| [openshift/openstack-cinder-csi-driver-operator](https://github.com/openshift/openstack-cinder-csi-driver-operator/)   | ose-openstack-cinder-csi-driver-operator.yml |
+| [openshift/csi-driver-manila-operator](https://github.com/openshift/csi-driver-manila-operator/)                       | csi-driver-manila-operator.yml               |
```

Comment:
Renaming `csi-driver-manila-operator` is orthogonal to this enhancement. It can be done in parallel, still, image names are hardcoded on so many places that it's basically impossible to rename them. There is [a guide](https://docs.ci.openshift.org/docs/how-tos/onboarding-a-new-component/#renaming-or-removing-components-in-the-openshift-release-payload) for changing the name of an image in OCP payload, but e.g. name in registry.redhat.io or in brew have a different process.

It might be better to start with a new image, i.e. start building `ose-openstack-manila-csi-driver-operator`, make sure it propagates everywhere, start using it and stop building `csi-driver-manila-operator` and deprecate / remove it from release. And build it only in z-streams, until they get EOL.

IIRC, the image name (+ repos) were requested by [Tim Bielawa](https://issues.redhat.com/browse/ART-2013), outside of our storage team and we did not review it :-(.